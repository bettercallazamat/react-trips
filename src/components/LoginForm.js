import PropTypes from 'prop-types';
import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { requestLogin } from '../helpers/requests';

const LoginForm = ({ onCancel }) => {
  const [userCred, setUserCred] = useState({ username: '', password: '' });
  const dispatch = useDispatch();

  let username;
  let password;

  const handleChange = (e) => {
    if (e.target.name === 'username') {
      username = e.target.value;
      setUserCred((state) => ({ ...state, username }));
    }
    if (e.target.name === 'password') {
      password = e.target.value;
      setUserCred((state) => ({ ...state, password }));
    }
  };

  const handleLogin = (e) => {
    e.preventDefault();
    requestLogin(dispatch, userCred.username, userCred.password);
    e.target.reset();
    onCancel();
  };

  return (
    <form onSubmit={handleLogin} className="LoginForm">
      <input name="username" type="text" onChange={handleChange} />
      <input name="password" type="password" onChange={handleChange} />
      <button type="submit">Login</button>
      <button type="button" onClick={onCancel}>Cancel</button>
    </form>
  );
};

LoginForm.propTypes = {
  onCancel: PropTypes.func.isRequired,
};

export default LoginForm;
