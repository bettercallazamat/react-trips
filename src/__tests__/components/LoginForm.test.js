import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import LoginForm from '../../components/LoginForm';

describe('LoginForm', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore([]);
  const store = mockStore({});
  store.dispatch = jest.fn();

  it('should match the snapshot', () => {
    const wrapper = shallow(<Provider store={store}><LoginForm onCancel={jest.fn} /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
