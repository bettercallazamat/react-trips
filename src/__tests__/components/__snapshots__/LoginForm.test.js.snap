// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `ShallowWrapper {}`;
